src/components/GamePage.vue

```diff
@@ -121,18 +123,63 @@ keys["Alt"] = "Alt";
 keys["Meta"] = "Meta";
 keys[" "] = "space";
 
-//キーボード押したらひかる
-//ロジックとしては押したボタンのkey.valueを取得する。"."+key.valueというクラスリストを持つノードを探す。（それぞれに対応するクラスの名前はあらかじめひつ一つのclasslistに書いてある。そのノードに光る要素を加える。keyupしたらremove。
+
+let index = 0;
+
 const KeyDown = () => {
+  if(code.finishcode.length === index){
+    console.log("finished");
+    return;
+  }
+  console.log(KeyboardEvent.keyCode)
+  //1: 最初にスペースを押したら始まる
+  if(code.correctcode === "" && event.key === " "){
+    for(let i = 0; i < code.fullcode.length; i++){
+        if(code.fullcode[i] === "<"){
+          code.correctcode = code.fullcode.substring(0,i)
+          code.pointercode = code.fullcode.substring(i,i+1);
+          code.notyetcode = code.fullcode.substring(i+1);
+          index = i;
+          break;
+      }
+    }
+  }
+  //2: ポインターと打ったキーが同じかどうかの判定
+  else if(event.key === code.pointercode){
+    //2-1: もしも">"だったら次の"<"までインデックスをスキップ
+    if(event.key === ">"){
+      code.correctcode = code.fullcode.substring(0,index);
+      //今のindex以降で次にある"<"
+      for(let i = index; i < code.fullcode.length; i++){
+        if(code.fullcode[i] === "<"){
+          code.correctcode = code.fullcode.substring(0,i)
+          code.pointercode = code.fullcode.substring(i,i+1);
+          code.notyetcode = code.fullcode.substring(i+1);
+          index = i;
+          //見つかり次第終了
+          break;
+        }
+      }
+    }
+    //2-2: もしも">"ではなく普通に一致したら次のindexへ
+    else{
+      code.correctcode = code.fullcode.substring(0,index)
+      code.pointercode = code.fullcode.substring(index,index+1);
+      code.notyetcode = code.fullcode.substring(index+1);
+      index += 1
+    }
+  }
+  //3: 間違っていた場合の処理
+  else{
+    console.log("you clicked wrong key");
+  }
   for(let key in keys){
       if(key === event.key){
         keyboard.value.querySelectorAll("." + keys[key])[0].classList.remove("bg-gray-100")
         keyboard.value.querySelectorAll("." + keys[key])[0].classList.add("bg-indigo-500")
       }
     }
```

Comment:
```suggestion
if (keys[event.key]) {
  keyboard.value.querySelectorAll("." + keys[event.key])[0]
    .classList.remove("bg-gray-100");
  keyboard.value.querySelectorAll("." + keys[event.key])[0]
    .classList.add("bg-indigo-500");
}
```
keyを指定するとvalueを取得できます
O(n) -> O(1)